### PR TITLE
fix(select): dynamic filter height fix

### PR DESCRIPTION
**Pull request by daanishnasir**
- fix console errors
- list will now shrink dynamically, no fixed height 
- no horizontal scroll bar 

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/lib/select/select.component.ts [hunk 1]
```diff
@@ -115,10 +114,9 @@ export class SelectChange {
 
       <div
         class="md-list md-list--vertical"
-        [style.height]="this.options.length > 8 ? scrollHeight : 'auto'"
-        [style.max-height]="scrollHeight"
+        [style.max-height]="this.options && this.options.length > 8 ? scrollHeight : '350px'"
         role="listbox"
-        style="align-items: normal; overflow: scroll;"
+        style="align-items: normal; overflow: scroll; overflow-x:hidden"
```
Comment (ArthusLiang) [inline angular/src/lib/select/select.component.ts]: Can we move this to core?

**Comment by daanishnasir** [comment]
PROVISION-5742

File: angular/src/lib/select/select.component.ts [hunk 2]
```diff
@@ -115,10 +114,9 @@ export class SelectChange {
 
       <div
         class="md-list md-list--vertical"
-        [style.height]="this.options.length > 8 ? scrollHeight : 'auto'"
-        [style.max-height]="scrollHeight"
+        [style.max-height]="this.options && this.options.length > 8 ? scrollHeight : '350px'"
```
Comment (mathsusan) [inline angular/src/lib/select/select.component.ts]: 350px?  where did that come from?

Comment (daanishnasir) [reply angular/src/lib/select/select.component.ts]: A default height was needed to avoid a large list showing

Comment (rovshenn) [reply angular/src/lib/select/select.component.ts]: Can we use rem instead of px to make sure it is a relative measure?

**Pull request merged**